src/cgame/cg_rocket_datasource.cpp

```diff
@@ -251,7 +251,11 @@ void CG_Rocket_BuildServerList()
 	// Only refresh once every 200 ms
 	if ( rocketInfo.realtime < 200 + rocketInfo.serversLastRefresh )
 	{
-		trap_LAN_UpdateVisiblePings( netSrc ); // keep the ping queue moving
+		if ( rocketInfo.data.serverListPopulated )
+		{
+			trap_LAN_UpdateVisiblePings( netSrc ); // keep the ping queue moving
+		}
+
 		return;
 	}
```

Comment:
Why not just move this code after line 274 so we don't need to add another bool to track state?

Reply:
I realized that my code had a regression in that there can be a delay up to 200 ms before UpdateVisiblePings is called for the first time, and rewrote it. I ended up getting rid of the bool and instead have a magic behavior when `rocketInfo.serversLastRefresh == rocketInfo.realtime`